### ISSUE TITLE
Video upload fills in the temp dir with files until virtual machine exits

### ISSUE THREAD

**Issue by lucaa** [apivideo/java-sdk#5]
How to reproduce:
* use the video upload to upload multiple files (larger than the chunk size of 128MB), in the same java process / instance
* in the temp folder, the sum of the sizes of the files will accumulate in the temporary folder until the java virtual machine is stopped (the process exits)

Expected result: the temp folder is cleaned up of all the data from the files that have been successfully uploaded.

This is particularly problematic on web applications, on which the java process ends **only when the web server is restarted**. 

The code here https://github.com/apivideo/java-sdk/blob/master/src/main/java/video/api/java/sdk/infrastructure/unirest/video/UnirestVideoClient.java#L140 chunks up the file to upload in pieces of 128MB in the temp folder, but only deletes them on the _exit of the virtual machine_ :  https://github.com/apivideo/java-sdk/blob/master/src/main/java/video/api/java/sdk/infrastructure/unirest/video/UnirestVideoClient.java#L167 . 

From deleteOnExit javadoc:
> Requests that the file or directory denoted by this abstract pathname be deleted when the virtual machine terminates.

**Comment by lucaa** [comment]
As a quick fix, the line on #167 should be replaced with:
```
                inputStream.close();
                chunkFile.delete();
```
if the requestExecutor.executeJson() function is synchronous, but it looks fairly synchronous to me.

The longer and more solid fix would be to also handle errors a little better in that code, and add the stream closing and file deletion in a finally{} block of that try, in order to avoid chunks remaining in temp folder upon upload error.

**Comment by dougsillars** [comment]
closing, fixed in today's merge, based on your code.  Thank you!
